### ISSUE TITLE
error message for malformed peer addresses is unhelpful

### ISSUE THREAD

**Issue by abuehrle** [weaveworks/weave#1892]
When I do a `weave launch http://172.17.8.102/`   the container dies.    but `weave launch 172.17.8.102` works fine. 

```
Command line options: map[port:6783 datapath:datapath dns-effective-listen-address:172.18.0.1 dns-listen-address:172.18.0.1:53 http-addr:127.0.0.1:6784 ipalloc-range:10.32.0.0/12 name:9e:e7:55:5a:ec:70 nickname:weave-microservice-02]
INFO: 2016/01/13 22:44:04.994004 Command line peers: [http://172.17.8.102/]
INFO: 2016/01/13 22:44:04.996229 Communication between peers is unencrypted.
INFO: 2016/01/13 22:44:04.999622 Our name is 9e:e7:55:5a:ec:70(weave-microservice-02)
INFO: 2016/01/13 22:44:05.016080 Docker API on unix:///var/run/docker.sock: &[Version=1.9.1 ApiVersion=1.21 GitCommit=a34a1d5 GoVersion=go1.4.3 Os=linux Arch=amd64 KernelVersion=3.13.0-68-generic BuildTime=Fri Nov 20 17:56:04 UTC 2015]
INFO: 2016/01/13 22:44:05.016128 Assuming quorum size of 2
INFO: 2016/01/13 22:44:05.016334 Listening for DNS queries on 172.18.0.1
INFO: 2016/01/13 22:44:05.016380 Sniffing traffic on datapath (via ODP)
FATA: 2016/01/13 22:44:05.016450 unknown port tcp///172.17.8.102/
```

**Comment by dpw** [comment]
The peers passed to `weave launch` must be hostnames or IP addresses (optionally with a `:<port number>` suffix).  URLs such as `http://172.17.8.102/` are not meaningful in that context.  (`http` gets interpreted as a hostname, and `//172.17.8.102/` as a port number, which leads to the "unknown port" error.)

**Comment by abuehrle** [comment]
Yes, I realized it was my mistake, but maybe it shouldn’t crash Weave. That’s all I meant by this report.

**Comment by rade** [comment]
*All* invalid arguments cause weave to stop. What else can it do? Besides reporting the problem more directly than the "Container has died; consult the logs." currently shown, which is the subject of #1280.

**Comment by abuehrle** [comment]
Maybe a better message would work, like catching the error and returning a more helpful message.

**Comment by rade** [comment]
So the reason it says "unknown port" rather than "invalid port" is that the port can be a service name, resolved via `/etc/services`, e.g. if you stuck an entry for weave in that you could write s.t. like "172.17.8.102:weave". Not that I can think of any reason why you'd want to do that. And since "service-names can be any printable characters excluding space and tab", there is preciously little validation that can be performed, e.g. "//172.17.8.102/" is actually a syntactically valid service name.

**Comment by awh** [comment]
Duplicate of #956?

**Comment by rade** [comment]
Not quite. The error message in #956 is kinda sensible. e.g. if I `weave launch unknown-host` it will say `lookup unknown-host: no such host`. The error message here is rather obscure.

**Comment by bboreham** [comment]
The fix in this repo is #1924
